### DISCUSSION TITLE
spawn must need to be detach()?

### DISCUSSION THREAD

**Discussion by amwps290** [longbridge/gpui-component#1486] [Q&A]
I'm learning how to use gpui and gpui-component, and I've encountered a problem. Does cx.spawn have to be detached? But I noticed that some code in gpui-component doesn't seem to use detach, and I don't quite understand why.

```
use gpui::{Task, *};
use gpui_component::{button::*, *};
use std::thread;
use std::time::Duration;

pub struct HelloWorld {}

impl Render for HelloWorld {
    fn render(&mut self, _: &mut Window, _: &mut Context<Self>) -> impl IntoElement {
        div()
            .v_flex()
            .gap_2()
            .size_full()
            .items_center()
            .justify_center()
            .child("Hello, World!")
            .child(
                Button::new("ok")
                    .primary()
                    .label("Let's Go!")
                    .on_click(|_, _, cx| {
                        let print_task = cx.spawn(async move |cx| println!("clicked"));
                    }),
            )
    }
}

fn main() {
    let app = Application::new();
    app.run(move |cx| {
        // This must be called before using any GPUI Component features.
        gpui_component::init(cx);

        cx.spawn(async move |cx| {
            cx.open_window(WindowOptions::default(), |window, cx| {
                let view = cx.new(|_| HelloWorld {});
                // This first level on the window, should be a Root.
                cx.new(|cx| Root::new(view.into(), window, cx))
            })?;

            Ok::<_, anyhow::Error>(())
        })
        .detach();
    });
}

```

in table_story.rs 

```

    fn load_more(&mut self, _: &mut Window, cx: &mut Context<Table<Self>>) {
        self.loading = true;

        self._load_task = cx.spawn(async move |view, cx| {
            // Simulate network request, delay 1s to load data.
            Timer::after(Duration::from_secs(1)).await;

            _ = cx.update(|cx| {
                let _ = view.update(cx, |view, _| {
                    view.delegate_mut().stocks.extend(random_stocks(200));
                    view.delegate_mut().loading = false;
                    view.delegate_mut().eof = view.delegate().stocks.len() >= 6000;
                });
            });
        });
    }
```

**Comment by huacnlee** [comment]
Yes.

There are 2 options:

1. Use `detach`.
2. Keep the `Task` in some owner.

Unless that, Rust will drop the ownerless variables, this is by Rust design, not about the GPUI.
